src/main/webapp/app/exercises/programming/manage/update/programming-exercise-update.component.ts

```diff
@@ -84,6 +84,9 @@ export class ProgrammingExerciseUpdateComponent implements OnInit {
     // with the restriction to a-z,A-Z,_ as "Java letter" and 0-9 as digits due to JavaScript/Browser Unicode character class limitations
     packageNamePatternForJavaKotlin =
         '^(?!.*(?:\\.|^)(?:abstract|continue|for|new|switch|assert|default|if|package|synchronized|boolean|do|goto|private|this|break|double|implements|protected|throw|byte|else|import|public|throws|case|enum|instanceof|return|transient|catch|extends|int|short|try|char|final|interface|static|void|class|finally|long|strictfp|volatile|const|float|native|super|while|_|true|false|null)(?:\\.|$))[A-Z_a-z][0-9A-Z_a-z]*(?:\\.[A-Z_a-z][0-9A-Z_a-z]*)*$';
+    // No dots allowed for the blackbox project type, because the folder naming works slightly different here.
```

Comment:
Here is another occurence of it. The regex is for the blackbox package name, so I'm not sure if blackbox project type is correct here in the comment.

Reply:
Also here: `No dots allowed in package names when the blackbox project type is selected`?

Reply:
But the variable name `packageNamePatternForJavaKotlin` already encodes that piece of information, so it should be fine here as is.

Reply:
Here is really optional, since it's just a comment. Feel free to leave it as it is